llvm/lib/MC/MCCASObjectV1.cpp

```diff
@@ -936,13 +937,83 @@ Error MCCASBuilder::createStringSection(
   return Error::success();
 }
 
+/// Reads and returns the length field of a debug info header contained in
+/// Reader, assuming Reader is positioned at the beginning of the section.
+/// The Reader's state is advanced to the first byte after the section.
+Expected<size_t> getSizeFromDebugHeaderAndSkip(BinaryStreamReader &Reader) {
+  // From Dwarf 5 section 7.4:
+  // In the 32-bit DWARF format, an initial length field [...] is an unsigned
+  // 4-byte integer (which must be less than 0xfffffff0);
+  uint32_t Word1;
+  if (auto E = Reader.readInteger(Word1))
+    return E;
+
+  // TODO: handle 64-bit DWARF format.
+  assert(Word1 < 0xfffffff0);
+
+  if (auto E = Reader.skip(Word1))
+    return E;
+  return Word1;
+}
+
+/// For each compile unit inside the __debug_info section, create a
+/// corresponding DebugInfoRef CAS object.
+Error MCCASBuilder::splitDebugInfoCUs() {
+  auto *DebugInfoSection =
+      Asm.getContext().getObjectFileInfo()->getDwarfInfoSection();
```

Comment:
I do not think this is needed, can you not just pass the Section from line 1012 to the function rather than get it again?

Reply:
I was debating about this too. The current implementation queries the object file twice, but the suggested approach gives the false illusion that the `splitDebugInfoCUs` function works with _any_ `MCSection`.

Thoughts?

Reply:
Oh I see what you mean, couldn't an explicit comment take care of that?

Reply:
It could indeed, but here's an idea: I see that, in many places both here and in my own PR, we constantly have to query for X or Y debug sections. I think we should have an NFC patch that creates a struct with pointers to all those sections and make such struct a member of `MCCASBuilder`, initialized at its constructor